source/connect/stable-api.txt

```diff
@@ -0,0 +1,114 @@
+.. _ruby-stable-api:
+
+===========
+Stable API
+===========
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: compatible, backwards, upgrade
+
+.. note::
+
+   The Stable API feature requires {+mdb-server+} 5.0 or later.
+
+Overview
+--------
+
+In this guide, you can learn how to specify **{+stable-api+}** compatibility when 
+connecting to a MongoDB deployment.
+
+The {+stable-api+} feature forces the server to run operations with behaviors compatible 
+with the API version you specify. Using the {+stable-api+} ensures consistent responses 
+from the server and provides long-term API stability for your application.
+
+The following sections describe how you can enable and customize {+stable-api+} for
+your MongoDB client. For more information about the {+stable-api+}, including a list of 
+the commands it supports, see :manual:`Stable API </reference/stable-api/>` in the
+{+mdb-server+} manual.
+
+Enable the {+stable-api+}
+-------------------------
+
+To enable the {+stable-api+}, pass a hash that specifies the {+stable-api+} version to the optional
+``server_api`` parameter when you create a ``Mongo::Client`` instance.
+
+The following code example shows how to specify {+stable-api+} version 1:
+
+.. code-block:: ruby
+
+   client = Mongo::Client.new(uri, server_api: { version: '1' })
+
+Once you create a ``Client`` instance with
+a specified API version, all commands that you run with the client use the specified
+version. If you must run commands using more than one version of the 
+{+stable-api+}, create a new ``Client``.
+
+Configure the {+stable-api+}
+----------------------------
+
+The following table describes {+stable-api+} options that you can set by specifying
+them in the ``server_api`` hash. You can use these options to customize the behavior of the
+{+stable-api+}.
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 25,75
+
+   * - Option Name
+     - Description
+
+   * - strict
+     - | **Optional**. When ``true``, if you call a command that isn't part of 
+         the declared API version, the driver raises an exception.
+       |
+       | Default: **false**
+
+   * -  deprecation_errors
+     - | **Optional**. When ``true``, if you call a command that is deprecated in the 
+         declared API version, the driver raises an exception.
+       |
+       | Default: **false**
+
+The following code example shows how you can set the two options on a ``ServerApi`` instance:
+
+.. code-block:: ruby
+
+   client = Mongo::Client.new(uri, 
+              server_api: { version: '1', strict: true, deprecation_errors: true })
+
+Troubleshooting
+---------------
+
```

Comment:
```suggestion
The following are common exceptions you may encounter:
```
S: I think an intro sentence would be nice here, so readers know what to expect.